### PR TITLE
Barista/icon overview page

### PR DESCRIPTION
**Pull request by lara-aigmueller**
### <strong>Pull Request</strong>

wait for #231 to be merged.

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/barista/src/components/lazy-icon/lazy-icon.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<a href="/resources/icons/{{ name }}" class="ba-lazy-icon-link">
+  <dt-icon [name]="name" *ngIf="_isVisible" class="ba-lazy-icon"></dt-icon>
```
Comment (thomaspink) [inline apps/barista/src/components/lazy-icon/lazy-icon.html]: The class name `ba-lazy-icon` should be only used for the host element.

File: apps/barista/src/components/lazy-icon/lazy-icon.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+:host {
+  min-height: 100px;
+}
+
+.ba-lazy-icon {
```
Comment (thomaspink) [inline apps/barista/src/components/lazy-icon/lazy-icon.scss]: As described above, pls change the class name

File: apps/barista/src/components/lazy-icon/lazy-icon.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+})
+export class BaLazyIcon {
+  @Input() name: string;
+  @Input() title: string;
+
+  /** @internal Whether the icon should be shown. */
+  _isVisible = false;
+
+  constructor(elementRef: ElementRef, changeDetectorRef: ChangeDetectorRef) {
+    createInViewportStream(elementRef, 0).subscribe(isInViewport => {
```
Comment (thomaspink) [inline apps/barista/src/components/lazy-icon/lazy-icon.ts]: Pls unsubscribe when component is destroyed

File: apps/barista/src/pages/page-outlet.ts [hunk 4]
```diff
@@ -29,10 +29,12 @@ import { BaPageMetaBase } from '@dynatrace/barista-components/barista-definition
 import { BaIndexPage } from './index-page/index-page';
 import { BaOverviewPage } from './overview-page/overview-page';
 import { BaSinglePage } from './single-page/single-page';
+import { BaIconOverviewPage } from './icon-overview-page/icon-overview-page';
 
 const LAYOUT_PAGES_MAPPING = {
   default: BaSinglePage,
   overview: BaOverviewPage,
+  iconoverview: BaIconOverviewPage,
```
Comment (thomaspink) [inline apps/barista/src/pages/page-outlet.ts]: Should we camelcase it for better readability?

File: libs/shared/barista-definitions/src/lib/barista-definitions.ts [hunk 5]
```diff
@@ -82,6 +84,21 @@ export interface BaIndexPageContent {
   cta: BaStrapiCTA;
 }
 
+/** Icon overview page */
+export interface BaIconOverviewPageContent {
```
Comment (thomaspink) [inline libs/shared/barista-definitions/src/lib/barista-definitions.ts]: A lot of interfaces have now the same subset of proerties (title, layout , ...). Can we create a common interface and extend from it?

Comment (lara-aigmueller) [reply libs/shared/barista-definitions/src/lib/barista-definitions.ts]: I would like to cleanup and organize all the interfaces in a separate step/PR after all other major issues have been resolved.

File: apps/barista/src/pages/icon-overview-page/icon-overview-page.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  /** Subscription on filter change event */
+  private _filterChangeSubscription = Subscription.EMPTY;
+
+  ngOnInit(): void {
+    this._updateFilteredIcons('');
+  }
+
+  ngAfterViewInit(): void {
+    this._filterChangeSubscription = fromEvent(
```
Comment (tomheller) [inline apps/barista/src/pages/icon-overview-page/icon-overview-page.ts]: You could use banana bracket value bindings with ngModel to get changing value inputs. 
You don't need to create a stream from the input event.
See: https://angular.io/api/forms/NgModel#using-ngmodel-on-a-standalone-control

File: apps/barista/src/pages/icon-overview-page/icon-overview-page.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    const nameMatchIcons = allIcons.filter(
+      icon => icon.name.toLowerCase().indexOf(filterValue) !== -1,
+    );
+    const tagMatchIcons = allIcons.filter(
+      icon =>
+        // Remove name matches to avoid duplicates in the result
+        !nameMatchIcons.includes(icon) &&
+        (!icon.tags ||
+          icon.tags.filter(
+            tagname => tagname && tagname.indexOf(filterValue) !== -1,
```
Comment (tomheller) [inline apps/barista/src/pages/icon-overview-page/icon-overview-page.ts]: Could you synchronize casing between tagname and filtervalue, to make searching case insensitive.

File: libs/shared/barista-definitions/src/lib/barista-definitions.ts (hunk 5 above)
Comment (thomaspink) [reply libs/shared/barista-definitions/src/lib/barista-definitions.ts]: Created an issue for this: #262

**Comment by thomaspink** [comment]
LGTM

**Pull request merged**